dgl_sparse/src/sparse_format.cc

```diff
@@ -0,0 +1,35 @@
+/**
+ *  Copyright (c) 2022 by Contributors
+ * @file sparse_format.cc
+ * @brief DGL C++ sparse format implementations.
+ */
+// clang-format off
+#include <sparse/dgl_headers.h>
+// clang-format on
+
+#include <sparse/sparse_format.h>
+
+#include "./utils.h"
+
+namespace dgl {
+namespace sparse {
+
+std::shared_ptr<COO> CSRToCOO(
+    int64_t num_rows, int64_t num_cols, const std::shared_ptr<CSR> csr) {
+  auto indptr = TorchTensorToDGLArray(csr->indptr);
+  auto indices = TorchTensorToDGLArray(csr->indices);
+  bool data_as_order = false;
+  runtime::NDArray data = aten::NullArray();
+  if (csr->value_indices.has_value()) {
+    data_as_order = true;
+    data = TorchTensorToDGLArray(csr->value_indices.value());
+  }
+  auto dgl_csr = aten::CSRMatrix(num_rows, num_cols, indptr, indices, data);
```

Comment:
* Perhaps we could add some util functions like `CSRToOldDGLCSR`, `OldDGLCSRToCSR` as they will be quite commonly used in other operators.
* I also suggest to add the `sorted` related flags to the new CSR/COO structure. They are performance critical for certain operations.

Reply:
OK. I will have these two things in the next PR. This PR is mainly for linking to DGL.